File: SafetyProgram.UI.Views/ModelViews/DocumentViews/IDocumentViewModel.cs
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Windows.Controls;
using ReactiveUI;

namespace SafetyProgram.UI.Views.ModelViews.DocumentViews
{
    /// <summary>
    /// Defines a viewmodel for a document view.
    /// </summary>
    public interface IDocumentViewModel : 
        INotifyPropertyChanged
    {
        decimal Width { get; }

        decimal Height { get; }

        /// <summary>
        /// Get the documentobjects in the document.
        /// </summary>
        IReactiveDerivedList<Control> DocumentObjects { get; }
    }
}

File: SafetyProgram.UI.ViewModels/IDocumentViewModel.fs
namespace SafetyProgram.UI.ViewModels

open System.Collections.ObjectModel
open System.ComponentModel
open System.Windows.Controls
open ReactiveUI

type IDocumentViewModel =

    inherit IViewModel
    
    abstract Width : decimal with get
    abstract Height : decimal with get
    abstract DocumentObjects : IReactiveDerivedList<Control> with get